lib/DxilPIXPasses/DxilAnnotateWithVirtualRegister.cpp

```diff
@@ -278,16 +336,18 @@ static uint32_t GetStructOffset(llvm::GetElementPtrInst *pGEP,
       MemberOffset += CountStructMembers(pStruct->getElementType(i));
     }
 
-    return MemberOffset + GetStructOffset(pGEP, GEPOperandIndex,
-                                          pStruct->getElementType(MemberIndex));
+    return AddConstIntValues(
+        m_DM->GetOP()->GetU32Const(MemberOffset),
+        GetStructOffset(pGEP, GEPOperandIndex,
+                        pStruct->getElementType(MemberIndex)));
   } else {
-    return 0;
+    return m_DM->GetOP()->GetU32Const(0);
   }
 }
 
 bool DxilAnnotateWithVirtualRegister::IsAllocaRegisterWrite(
-    llvm::Value *V, llvm::AllocaInst **pAI, llvm::Value **pIdx) {
-  llvm::IRBuilder<> B(m_DM->GetCtx());
+    hlsl::OP *HlslOP, llvm::Value *V, llvm::AllocaInst **pAI,
```

Comment:
Is `HlslOP` here just for getting Constant Int? You could just use `B.getInt32()` below. If that `IRBuilder` is not used for anything other than getting the constant, you can also just do `ConstantInt::get` directly.

Reply:
We could just get rid of the HlslOP arg in both `SplitVectorStores` and `IsAllocaRegisterWrite`.